examples/3d/3d_generate_custom_mesh.rs

```diff
@@ -0,0 +1,245 @@
+        // .add_systems(Update, rotate_z)
+        .add_systems(Update, input_handler)
+        .run();
+}
+
+// Rotate the cube to show off all the sides.
+#[allow(dead_code)]
+fn rotate_y(mut query: Query<&mut Transform, With<Handle<Mesh>>>, time: Res<Time>) {
+    for mut transform in &mut query {
+        transform.rotate_y(time.delta_seconds() / 1.2);
+    }
+}
+
+// Optional system to rotate around the x axis.
+#[allow(dead_code)]
+fn rotate_x(mut query: Query<&mut Transform, With<Handle<Mesh>>>, time: Res<Time>) {
+    for mut transform in &mut query {
+        transform.rotate_x(time.delta_seconds() / 1.2);
+    }
+}
+
+// Optional system to rotate around the z axis.
+#[allow(dead_code)]
+fn rotate_z(mut query: Query<&mut Transform, With<Handle<Mesh>>>, time: Res<Time>) {
+    for mut transform in &mut query {
+        transform.rotate_z(time.delta_seconds() / 1.2);
+    }
+}
+
+fn setup(
+    mut commands: Commands,
+    asset_server: ResMut<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+) {
+    // Importing the custom texture.
+    let custom_texture_handle: Handle<Image> = asset_server.load("textures/array_texture.png");
+    // Creating and saving a handle to the mesh.
+    let cube_mesh_handle: Handle<Mesh> = meshes.add(create_cube_mesh());
+
+    // Rendering the mesh with the custom texture using a PbrBundle.
+    commands.spawn(PbrBundle {
+        mesh: cube_mesh_handle,
+        material: materials.add(StandardMaterial {
+            base_color_texture: Some(custom_texture_handle),
+            ..default()
+        }),
+        ..default()
+    });
+
+    // Transform for the camera and lighting, looking at (0,0,0) (the position of the mesh).
+    let camera_and_light_transform =
+        Transform::from_xyz(3.0, 3.0, 3.0).looking_at(Vec3::ZERO, Vec3::Y);
+
+    // Camera in 3D space.
+    commands.spawn(Camera3dBundle {
+        transform: camera_and_light_transform,
+        ..default()
+    });
+
+    // Lighting up the scene.
+    commands.spawn(PointLightBundle {
+        point_light: PointLight {
+            intensity: 9000.0,
+            range: 100.0,
+            ..default()
+        },
+        transform: camera_and_light_transform,
+        ..default()
+    });
+}
+
+// System to recieve input from the user, check out examples/input/ for more examples about user
+// input.
+fn input_handler(
+    keyboard_input: Res<Input<KeyCode>>,
+    mesh_query: Query<&Handle<Mesh>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+) {
+    if keyboard_input.just_pressed(KeyCode::Space) {
+        let mesh_handle = mesh_query.get_single().expect("Query not successful");
+        let mesh = meshes.get_mut(mesh_handle).unwrap();
+        toggle_texture(mesh);
+    }
+}
+
+fn create_cube_mesh() -> Mesh {
+    let mut cube_mesh = Mesh::new(PrimitiveTopology::TriangleList);
+
+    #[rustfmt::skip]
+    cube_mesh.insert_attribute(
+        Mesh::ATTRIBUTE_POSITION,
```

Comment:
We should probably have a comment about winding direction somewhere around here.